www-apps/radicale/radicale-2.1.8.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="6"
+
+PYTHON_COMPAT=( python{3_4,3_5,3_6} )
+
+inherit eutils distutils-r1 user
+
+MY_PN="Radicale"
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A simple CalDAV calendar server"
+HOMEPAGE="http://radicale.org/"
+SRC_URI="mirror://pypi/${MY_PN:0:1}/${MY_PN}/${MY_P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="+bcrypt"
+
+RDEPEND=">=dev-python/vobject-0.9.5[${PYTHON_USEDEP}]
+	bcrypt? ( dev-python/passlib[bcrypt,${PYTHON_USEDEP}] )"
+
+S=${WORKDIR}/${MY_P}
+
+RDIR=/var/lib/radicale
+LDIR=/var/log/radicale
+
+pkg_setup() {
+	enewgroup radicale
+	enewuser radicale -1 -1 ${RDIR} radicale
+}
+
+python_install_all() {
+	rm README* || die
+
+	# init file
+	newinitd "${FILESDIR}"/radicale.init.d radicale
+
+	# directories
+	diropts -m0750
+	dodir ${RDIR}
+	fowners radicale:radicale ${RDIR}
+	keepdir ${RDIR}
+	diropts -m0755
+	dodir ${LDIR}
+	fowners radicale:radicale ${LDIR}
+	keepdir ${LDIR}
+
+	# config file
+	insinto /etc/${PN}
+	doins config logging
+
+	# fcgi and wsgi files
+	exeinto /usr/share/${PN}
+	doexe radicale.wsgi
+	doexe radicale.fcgi
+
+	distutils-r1_python_install_all
+}
+
+pkg_postinst() {
+	einfo "A sample WSGI script has been put into ${ROOT}usr/share/${PN}."
+	einfo "You will also find there an example FastCGI script."
+	ewarn "If you were previously using Radicale 1, manual migration is needed."
+	ewarn "Please see <http://radicale.org/1to2/> for details."
+}
```

Comment:
as mentioned in a comment in the PR, this should probably happen earlier so people can migrate data before merging

Reply:
That sounds a lot like a news item. I don’t see a Github mirror of `gentoo-news.git`; do you know if there is one or if there is another way for a non-developer to get news items added? Particularly, how to make that happen at about the same time as the new ebuild is committed?

Reply:
Sorry, no clue. I guess it can stay there and be a little more verbose. If the migration indeed requires 1.1.6 the ewarn should instruct people to downgrade again.

We need to make sure that:
- 2.x does not corrupt/delete old data so it can not be migrated anymore
- 1.1.6 stays in the tree for a long time

Or maybe, what we need can be implemented with blockers or slots, maybe in combination with a migration USE-flag.

Reply:
I’ve requested review on the gentoo-dev mailing list of a news item to be shown to all <radicale-2 users.

Reply:
Does the migration require using tools installed by v2 or is it done somehow else?

Reply:
`einfo` in ebuild removed in f94c56be68e4d05b3f14870e2159b5a997efd599 due to news item being more appropriate. @mgorny, to answer your question, to migrate, you must have v1 installed, then you run v1 with `--export-storage`, then you upgrade to v2, then you run v2 and use the directory that `--export-storage` created as your storage location for v2. So no special tools outside Radicale itself, but you do need to run `--export-storage` while you still have v1 installed.